src/bsp/src/stm32/hal/src/hal.c

```diff
@@ -14,6 +14,10 @@
 #include <lwip/apps/lwiperf.h>
 #endif
 
+#ifdef RUNTIME_STATS
+uint32_t HAL_getCPUCounter() { return RUNTIME_STATS_TIMER->Instance->CNT; }
```

Comment:
Not sure if this should go into a platform specific file, because the CNT register might not be named the same between architectures